.github/workflows/reusable-performance.yml

```diff
@@ -109,6 +110,11 @@ jobs:
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
         with:
           show-progress: ${{ runner.debug == '1' && 'true' || 'false' }}
+          fetch-depth: ${{ github.event_name == 'workflow_dispatch' && '2' || '1' }}
+
+      - name: Retrieve previous commit SHA (if necessary)
+        if: ${{ github.event_name == 'workflow_dispatch' }}
+        run: echo "TARGET_SHA=$(git rev-parse HEAD^1)" >> $GITHUB_ENV
```

Comment:
As long as these workflow_dispatch events are only happening on synced branches (i.e., trunk and release branches) this should be fine. If this were run against a PR branch with multiple commits, then I think the performance tests would compare the branch HEAD commit to the penultimate commit from the PR branch, which is not the intent.

Reply:
That makes sense. I will add a clarifying comment.

Because of the values supplied through contexts and event payloads, the conditions above where `TARGET_SHA` is defined at the workflow level should always be sufficient for determining this. If we add more events other than `push`, `pull_request`, and `workflow_dispatch`, then we may need to either expand this step or add conditions to the first location.